src/main/java/echoserver/ClientConnection.java

```diff
@@ -0,0 +1,40 @@
+package echoserver;
+
+import java.io.*;
+import java.net.Socket;
+
+public class ClientConnection {
+    public ClientConnection(){};
+
+    public ClientConnection(Socket socket) {
+        this.socket = socket;
+    }
+
+    public String promptInput() throws IOException {
+        lastSavedInput = convertInputStreamToString(socket.getInputStream());
+        return lastSavedInput;
+    }
+
+    public String readLastSavedInput(){
+        return lastSavedInput;
+    }
+
+    public OutputStream getOutputStream() throws IOException{
+        return socket.getOutputStream();
+    }
+
+    public Boolean isClosed(){
+       return socket.isClosed();
+    }
+
+    private
```

Comment:
Hm, I thought in Java you had to specify `private` in front of each private attribute or method? Is that not the case anymore?

Reply:
It looks like it changed so you don't have to do that anymore.